rmw_zenoh_cpp/src/detail/rmw_subscription_data.cpp

```diff
@@ -40,61 +40,23 @@
 
 namespace rmw_zenoh_cpp
 {
-namespace
-{
-//==============================================================================
-// TODO(Yadunund): Make this a class member and lambda capture weak_from_this()
-// instead of passing a rawptr to SubscriptionData when we switch to zenoh-cpp.
-void sub_data_handler(z_loaned_sample_t * sample, void * data)
```

Comment:
I notice that you've implemented the TODO below but I believe we wanted to do this in a follow up PR after we port the code to zenoh-cpp first. So for now, can we keep these functions and rely on `std::bind` or lambda to call these functions?

Reply:
I implemented the capture lambdas, implemening the lambdas the code is more easy to follow

Reply:
I think this one is OK to leave as well; we haven't had too much trouble with this one.

Reply:
We decided to stick with the lambdas, so resolving this comment.